server/remote/github/github.go

```diff
@@ -491,6 +493,53 @@ func (c *client) Branches(ctx context.Context, u *model.User, r *model.Repo) ([]
 
 // Hook parses the post-commit hook from the Request body
 // and returns the required data in a standard format.
-func (c *client) Hook(r *http.Request) (*model.Repo, *model.Build, error) {
-	return parseHook(r, c.MergeRef)
+func (c *client) Hook(ctx context.Context, r *http.Request) (*model.Repo, *model.Build, error) {
+	pull, repo, build, err := parseHook(r, c.MergeRef, c.PrivateMode)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if pull != nil && len(build.ChangedFiles) == 0 {
+		build, err = c.loadChangedFilesFromPullRequest(ctx, pull, repo, build)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	return repo, build, nil
+}
+
+func (c *client) loadChangedFilesFromPullRequest(ctx context.Context, pull *github.PullRequest, tmpRepo *model.Repo, build *model.Build) (*model.Build, error) {
+	_store, ok := store.TryFromContext(ctx)
+	if !ok {
+		return build, nil
```

Comment:
We should handle this error.

Reply:
how do you like to handle it, to be specific?

Reply:
log it?